### DISCUSSION TITLE
How to disable "terminal width 511" and "terminal length 0" which are sent by default?

### DISCUSSION THREAD

**Discussion by amitsova** [ktbyers/netmiko#3154] [Q&A] [answered]
Dear all,

Is there any chance to disable the "terminal width 511" and "terminal length 0" which are sent by default when using the send_multiline function?

From the netmiko log:

INFO:paramiko.transport:Authentication (password) successful!
DEBUG:paramiko.transport:[chan 0] Max packet in: 32768 bytes
DEBUG:paramiko.transport:[chan 0] Max packet out: 4096 bytes
DEBUG:paramiko.transport:Secsh channel 0 opened.
DEBUG:paramiko.transport:[chan 0] Sesch channel 0 request ok
DEBUG:paramiko.transport:[chan 0] Sesch channel 0 request ok
DEBUG:netmiko:write_channel: b'\n'
**DEBUG:netmiko:write_channel: b'terminal width 511\n'**
DEBUG:netmiko:read_channel: 
LKN-BTJ-01#
LKN-BTJ-01#
**DEBUG:netmiko:read_channel: terminal
DEBUG:netmiko:read_channel:  width 51
DEBUG:netmiko:read_channel: 1**

DEBUG:netmiko:Pattern found: (terminal width 511) 
LKN-BTJ-01#
LKN-BTJ-01#terminal width 511
DEBUG:netmiko:In disable_paging
DEBUG:netmiko:Command: terminal length 0

**DEBUG:netmiko:write_channel: b'terminal length 0\n'**
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
**DEBUG:netmiko:read_channel: LKN-BTJ-01#ter
DEBUG:netmiko:read_channel: minal lengt
DEBUG:netmiko:read_channel: h 0

DEBUG:netmiko:Pattern found: (terminal\ length\ 0) 
LKN-BTJ-01#terminal length 0
DEBUG:netmiko:
LKN-BTJ-01#terminal length 0
DEBUG:netmiko:Exiting disable_paging**
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:Clear buffer detects data in the channel
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:write_channel: b'\n'
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 
DEBUG:netmiko:read_channel: 

I really don't need these extra commands as I am trying some specific tests on Cisco devices. Please help me to workaround it!

Thank you in advance!

**Comment by ktbyers** [comment]
You can use the `generic` driver. It doesn't do anything post login (but it also generally won't have any platform specific behaviors so it might cause you issues in other places).

**Reply by amitsova** [reply]
Hi Kirk,

Thanks for the quick response! 

I was able to workaround it by commenting out the disable_paging and the set_terminal_width definitions in the base_connection.py and in the cisco_ios.py. Not the smartest solution, though :)

What is this generic driver? Would you please share an example? 

Thanks,
Adriana

**Comment by ktbyers** [comment, marked as answer]
Just use `device_type="generic"`.

It will make an SSH connection and do nothing else post login.

You could also make your own class and inherit from the Netmiko class and overwrite the session_preparation method. If you wanted to retain the other methods, but change the session_preparation behavior. Of course, you can just directly edit the Netmiko source code as well.

**Reply by amitsova** [reply]
I've just found an old discussion with you - https://github.com/ktbyers/netmiko/issues/909
And used:

`dev_info = {'device_type': 'generic_termserver_ssh', 'ip': i, 'username': 'kmp_ro', 'password': '*****'}`

Works perfectly fine for my tests! Thanks a ton!!!!

Adriana

**Reply by ktbyers** [reply]
Yeah, `generic` and `generic_termserver_ssh` are ultimately the same thing (i.e. point to the same class)
